src/python/pants/backend/awslambda/python/target_types.py

```diff
@@ -190,6 +191,7 @@ class PythonAWSLambda(Target):
     core_fields = (
         *COMMON_TARGET_FIELDS,
         OutputPathField,
+        InterpreterConstraintsField,
```

Comment:
Why was this field added? The PythonAwsLambdaRuntime says what runtime this is bound for. If some internal rule needs an IC, it should derive one from the PythonAwsLambdaRuntime. Otherwise all this field can do is be wrong and conflict with the designated PythonAwsLambdaRuntime.

Reply:
Oof, you are 100% right. My bad. This field is never consumed. The `PexFromTargets` does calculate the interpreter constraints of the transitive closure, but then discards it because platforms is set:

https://github.com/pantsbuild/pants/blob/017e4e5dacd5320f96f349444759721868ea3719/src/python/pants/backend/awslambda/python/rules.py#L78-L93

https://github.com/pantsbuild/pants/blob/017e4e5dacd5320f96f349444759721868ea3719/src/python/pants/backend/python/util_rules/pex.py#L315-L330

The other PEX we build looks at lambdex subsystem:

https://github.com/pantsbuild/pants/blob/017e4e5dacd5320f96f349444759721868ea3719/src/python/pants/backend/awslambda/python/rules.py#L95-L101

Will deprecate.